src/lib/JsResolver/JsResolverSecrets.ts

```diff
@@ -0,0 +1,111 @@
+import { Signer } from "@ethersproject/abstract-signer";
+import axios, { Axios } from "axios";
+import { getAuthToken, SiweOverride } from "./authToken";
+import { OPS_USER_API } from "../../constants";
+import { Secrets } from "../../types";
+import { errorMessage } from "../../utils";
+
+export class JsResolverSecrets {
+  private readonly _signer: Signer;
+  private readonly _userApi: Axios;
+
+  constructor(signer: Signer) {
+    this._signer = signer;
+    this._userApi = axios.create({
+      baseURL: OPS_USER_API,
+    });
+  }
+
+  public async get(key: string, override?: SiweOverride): Promise<string> {
+    try {
+      const address = await this._signer.getAddress();
+      const authToken = await getAuthToken(this._signer, override);
```

Comment:
also see how we store the signature in memory for GelatoOpsSDK to avoid asking/generating the signature again for every requests

Reply:
[c7685a1](https://github.com/gelatodigital/ops-sdk/pull/25/commits/c7685a1bbd659e487b8598900f90b37794006ae5)